### PR TITLE
Benchmarks: add ECDSA benchmarks

### PR DESCRIPTION
**Pull request by vkrasnov**

### CODE REVIEW

File: bench/ecdsa.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+}
+
+fn verify(c: &mut Criterion) {
+    for (alg_name, sign_alg, verify_alg) in ALGORITHMS {
+        c.bench_function(&format!("ecdsa_{alg_name}_verify"), |b| {
+            let rng = rand::SystemRandom::new();
+            let pkcs8_bytes = EcdsaKeyPair::generate_pkcs8(sign_alg, &rng).unwrap();
+            let key_pair = EcdsaKeyPair::from_pkcs8(sign_alg, pkcs8_bytes.as_ref(), &rng).unwrap();
+            let signature = key_pair.sign(&rng, &[]).unwrap();
```
Comment (briansmith) [inline bench/ecdsa.rs:55]: We're going to run into issue described in https://github.com/briansmith/ring/issues/1772 where the performance of each run is going to vary based on the randomized values of `r` and `s` in this signature.

File: bench/ecdsa.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    (
+        "p384",
+        &signature::ECDSA_P384_SHA384_FIXED_SIGNING,
+        &signature::ECDSA_P384_SHA384_FIXED,
+    ),
+];
+
+fn sign(c: &mut Criterion) {
+    for (alg_name, alg, _) in ALGORITHMS {
+        c.bench_function(&format!("ecdsa_{alg_name}_sign"), |b| {
```
Comment (briansmith) [inline bench/ecdsa.rs:37]: We're going to end up with the "fixed" and non-fixed (ASN.1) benchmarks having the same benchmark name.

I suggest we just remove the `_FIXED_` variants from the bemchmarking to resolve this. I don't think we need to worry about the difference in performance between the two ways of formatting the signature and also the performance difference between the two is probably impossible to measure in a meaningful way.

File: bench/ecdsa.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            b.iter(|| {
+                key_pair.sign(&rng, black_box(&[])).unwrap();
+            })
+        });
+    }
+}
+
+fn verify(c: &mut Criterion) {
+    for (alg_name, sign_alg, verify_alg) in ALGORITHMS {
+        c.bench_function(&format!("ecdsa_{alg_name}_verify"), |b| {
```
Comment (briansmith) [inline bench/ecdsa.rs:51]: ditto here.

**Review by briansmith — CHANGES_REQUESTED** [review]

File: bench/ecdsa.rs (hunk 2 above)
Comment (vkrasnov) [reply bench/ecdsa.rs:37]: Not sure what do you mean, the name is whatever string I put there in the ALGORITHMS array. I also think padding performance is not critical, so I only put the FIXED variants.

Comment (briansmith) [reply bench/ecdsa.rs:37]: I don't remember what it looked like before but it is fine the way it is now.

**Pull request merged**